src/service.py

```diff
@@ -55,13 +56,20 @@ def __init__(self, search_path: Path):
         self.config_template = self.environment.get_template(EXPORTER_CONFIG_TEMPLATE)
         self.service_template = self.environment.get_template(EXPORTER_SERVICE_TEMPLATE)
 
-    def _install(self, path: Path, content: str) -> bool:
+    def _install(self, path: Path, content: str, mode: Optional[int] = None) -> bool:
         """Install file."""
         success = True
         try:
             logger.info("Writing file to %s.", path)
-            with open(path, "w", encoding="utf-8") as file:
-                file.write(content)
+            if mode:
+                with os.fdopen(
+                    os.open(path, os.O_CREAT | os.O_WRONLY, mode), "w", encoding="utf-8"
+                ) as file:
+                    file.write(content)
+            else:
+                # create file with default permissions based on default OS umask
+                with open(path, "w", encoding="utf-8") as file:
+                    file.write(content)
```

Comment:
I think you could make this slightly more dry
```suggestion
            fileobj = (
                os.fdopen(
                    os.open(path, os.O_CREAT | os.O_WRONLY, mode), "w", encoding="utf-8"
                )
                if mode
                # create file with default permissions based on default OS umask
                else open(path, "w", encoding="utf-8")
            )
            with open(path, "w", encoding="utf-8") as file:
                file.write(content)
```

Reply:
I did have to disable a warning from pylint asking me to not use `open` without a context manager but in our case, I think it's cleaner overall (and we're ensuring the file object does get closed later).